bd-log-primitives/src/lib.rs

```diff
@@ -215,38 +252,239 @@ pub enum LogFieldKind {
 // Log
 //
 
-/// A copy of an incoming log line.
+// This is a wrapper around an owned log entry that is being processed by the workflow engine.
+// Most of this code is a custom protobuf serialization implementation to avoid the overhead of
+// the normal protobuf generated code which requires extensive copying. Note that the proto
+// implementation calculates size every time it's invoked vs. caching it. In general we expect
+// the size calculation to be fast and in the common case it is only computed once making
+// caching not useful.
 #[derive(Debug, PartialEq, Eq)]
 pub struct Log {
```

Comment:
It would be a bigger change but would it benefit us to construct the protobuf object early on and put it on the queue instead of constructing a Log then later on having to convert it to the proto?